solutions/silver/cf-1368D.mdx

```diff
@@ -0,0 +1,139 @@
+---
+id: cf-1368D
+source: CF
+title: AND, OR and square sum
+author: Paul Chen
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/79027)
+
+# Explanation
+
+**Observation 1**: Notice that $x \ AND \ y + x \ OR \ y = x + y$, so the total sum of all the elements never changes.
+
+**Observation 2**: Notice that during each operation, the number and position of the set bits don't change. For instance, let $x = 110_2$, $y = 101_2$. $x \ AND \ y = 100_2$, and $x \ OR \ y = 111_2$. Before and after the operation, there are two $1$ bits that are set, one $2$ bit that is set, and one $4$ bit that is set.
+
+**Observation 3**: Notice when performing an operation, $\sum_{i=1}^{n}{a_i^2}$ increases when $|x - y|$ increases. Thus, we want to keep performing operations such that $|x - y|$ increases. In fact, because $x \ AND \ y \leq \min(x, y)$ and $x \ OR \ y \geq \max(x, y)$, performing an operation will never decrease $|x - y|$.
+
+Therefore, the optimal configuration occurs when it is impossible to change $|x - y|$ with more operations. This happens when, for any two values $a_i$ and $a_j$, $a_i \ AND \ a_j = \min(a_i, a_j)$, and $a_i \ OR \ a_j = \max(a_i, a_j)$. In other words, when we sort the optimal configuration in ascending order, $a_i \ AND \ a_{i + 1} = a_i$.
+
+The optimal configuration can be built by handling each bit separately. This approach takes into account Observation 2, which states that no operations will alter the bits. For instance, consider the array, `[1, 5, 4, 7, 3]`. In binary form, it is represented as `[001, 101, 100, 111, 011]`. To structure this array optimally, move all the set bits to one end. This results in the reconfigured array: `[000, 001, 101, 111, 111]`. In decimal form, this new array reads `[0, 1, 5, 7, 7]`. Finally, find the sum of squares of the elements in this new array. The resulting total provides the desired solution.
```

Comment:
> To structure this array optimally, move all the set bits to one end.

What does set bits mean and also explain exactly how the array is reconfigured given you provided an example

Reply:
```suggestion
The optimal configuration can be built by handling each bit separately. This approach takes into account Observation 2, which states that no operations will alter the bits. For example, consider the array, `[1, 5, 4, 7, 3]`. In binary form, it is represented as `[001, 101, 100, 111, 011]`. To structure this array optimally, move all the set bits to one end. This results in the reconfigured array: `[000, 001, 101, 111, 111]`. In decimal form, this new array reads `[0, 1, 5, 7, 7]`. Finally, find the sum of squares of the elements in this new array. The resulting total provides the desired solution.
```